### PR TITLE
USHIFT-1590: skip failing invariants in MicroShift

### PR DESCRIPTION
**Pull request by pacevedom**

### CODE REVIEW

File: pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go [hunk 1]
```diff
@@ -51,9 +58,18 @@ func (w *newAPIServerDisruptionChecker) WriteContentToStorage(ctx context.Contex
...
-
+	isMicroShift, err := exutil.IsMicroShiftCluster(kubeClient)
+	if err != nil {
+		return err
+	}
+	if !isMicroShift {
+		if err := sampler.TearDownInClusterMonitors(w.adminRESTConfig); err != nil {
+			return err
+		}
+	}
```
Comment (deads2k) [inline pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go:73]:
```suggestion
	if isMicroShift {
	  return nil
	}
	if err := sampler.TearDownInClusterMonitors(w.adminRESTConfig); err != nil {
		return err
	}
```

File: pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go [hunk 2]
```diff
@@ -32,7 +33,13 @@ func (w *newAPIServerDisruptionChecker) CollectData(ctx context.Context, storage
 	if err != nil {
 		return nil, nil, err
 	}
-
+	isMicroShift, err := exutil.IsMicroShiftCluster(kubeClient)
+	if err != nil {
+		return nil, nil, err
+	}
+	if isMicroShift {
+		return nil, nil, nil
```
Comment (neisw) [inline pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go:41]: Wondering if there is value to carrying over the notSupportedReason like in https://github.com/pacevedom/origin/blob/ecdb4fbd1761c42b0eab8c0e41a4085aaa1ebb14/pkg/monitortests/network/disruptionserviceloadbalancer/monitortest.go#L84 

and then a check for it in https://github.com/pacevedom/origin/blob/ecdb4fbd1761c42b0eab8c0e41a4085aaa1ebb14/pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go#L69

File: pkg/monitortests/network/legacynetworkmonitortests/networking.go [hunk 3]
```diff
@@ -59,15 +61,26 @@ func testPodSandboxCreation(events monitorapi.Intervals, clientConfig *rest.Conf
...
 	var platform configv1.PlatformType
-	configClient, err := configclient.NewForConfig(clientConfig)
+
+	kubeClient, err := kubernetes.NewForConfig(clientConfig)
 	if err != nil {
-		failures = append(failures, fmt.Sprintf("error creating configClient: %v", err))
+		failures = append(failures, fmt.Sprintf("error creating kubeClient: %v", err))
 	} else {
-		infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+		isMicroShift, err := exutil.IsMicroShiftCluster(kubeClient)
```
Comment (deads2k) [inline pkg/monitortests/network/legacynetworkmonitortests/networking.go:69]: use early returns to eliminate nesting.  This is hard to follow.

File: pkg/monitortests/node/kubeletlogcollector/monitortest.go [hunk 4]
```diff
@@ -30,6 +31,15 @@ func (w *kubeletLogCollector) CollectData(ctx context.Context, storageDir string
 	if err != nil {
 		return nil, nil, err
 	}
+	// MicroShift does not have a proper journal for the node logs api.
```
[the remark this answers is not in the document]
Comment (pacevedom) [reply pkg/monitortests/node/kubeletlogcollector/monitortest.go:34]: Its under a different service, the journal belongs to microshift which holds the whole control plane logging output, not just kubelet. All the supportability docs are prepared for it (as well as an sos plugin)

File: pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go (hunk 2 above)
Comment (pacevedom) [reply pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go:47]: Done

File: pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go (hunk 1 above)
Comment (pacevedom) [reply pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go]: Done

File: pkg/monitortests/network/legacynetworkmonitortests/networking.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (pacevedom) [reply pkg/monitortests/network/legacynetworkmonitortests/networking.go]: Done

File: pkg/monitortests/node/kubeletlogcollector/monitortest.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (pacevedom) [reply pkg/monitortests/node/kubeletlogcollector/monitortest.go:34]: This is the output:
```
$ oc get --raw "/api/v1/nodes/microshift-dev/proxy/logs"
<pre>
<a href="README">README</a>
<a href="anaconda/">anaconda/</a>
<a href="audit/">audit/</a>
<a href="btmp">btmp</a>
<a href="btmp-20230810">btmp-20230810</a>
<a href="chrony/">chrony/</a>
<a href="containers/">containers/</a>
<a href="crio/">crio/</a>
<a href="cron">cron</a>
<a href="cron-20230704">cron-20230704</a>
<a href="cron-20230810">cron-20230810</a>
<a href="cron-20230817">cron-20230817</a>
<a href="cron-20230824">cron-20230824</a>
<a href="dnf.librepo.log">dnf.librepo.log</a>
<a href="dnf.log">dnf.log</a>
<a href="dnf.log.1">dnf.log.1</a>
<a href="dnf.rpm.log">dnf.rpm.log</a>
<a href="firewalld">firewalld</a>
<a href="hawkey.log">hawkey.log</a>
<a href="hawkey.log-20230704">hawkey.log-20230704</a>
<a href="hawkey.log-20230810">hawkey.log-20230810</a>
<a href="hawkey.log-20230817">hawkey.log-20230817</a>
<a href="hawkey.log-20230824">hawkey.log-20230824</a>
<a href="insights-client/">insights-client/</a>
<a href="kdump.log">kdump.log</a>
<a href="kube-apiserver/">kube-apiserver/</a>
<a href="lastlog">lastlog</a>
<a href="libvirt/">libvirt/</a>
<a href="maillog">maillog</a>
<a href="maillog-20230704">maillog-20230704</a>
<a href="maillog-20230810">maillog-20230810</a>
<a href="maillog-20230817">maillog-20230817</a>
<a href="maillog-20230824">maillog-20230824</a>
<a href="messages">messages</a>
<a href="messages-20230704">messages-20230704</a>
<a href="messages-20230810">messages-20230810</a>
<a href="messages-20230817">messages-20230817</a>
<a href="messages-20230824">messages-20230824</a>
<a href="openvswitch/">openvswitch/</a>
<a href="ovn/">ovn/</a>
<a href="ovn-kubernetes/">ovn-kubernetes/</a>
<a href="pods/">pods/</a>
<a href="private/">private/</a>
<a href="qemu-ga/">qemu-ga/</a>
<a href="rhsm/">rhsm/</a>
<a href="secure">secure</a>
<a href="secure-20230704">secure-20230704</a>
<a href="secure-20230810">secure-20230810</a>
<a href="secure-20230817">secure-20230817</a>
<a href="secure-20230824">secure-20230824</a>
<a href="spooler">spooler</a>
<a href="spooler-20230704">spooler-20230704</a>
<a href="spooler-20230810">spooler-20230810</a>
<a href="spooler-20230817">spooler-20230817</a>
<a href="spooler-20230824">spooler-20230824</a>
<a href="sssd/">sssd/</a>
<a href="swtpm/">swtpm/</a>
<a href="tallylog">tallylog</a>
<a href="wtmp">wtmp</a>
</pre>
```
Test is looking for _journal_ and then filters by the systemd unit for kubelet. In microshift the kubelet is part of microshift systemd unit.

File: pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go [hunk 5]
```diff
@@ -51,9 +68,19 @@ func (w *newAPIServerDisruptionChecker) WriteContentToStorage(ctx context.Contex
 }
 
 func (w *newAPIServerDisruptionChecker) Cleanup(ctx context.Context) error {
```
Comment (neisw) [inline pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go:70]: if len(w.notSupportedReason) > 0 {
		return nil
	}
	
?

Comment (pacevedom) [reply pkg/monitortests/kubeapiserver/disruptionnewapiserver/monitortest.go:70]: Done

File: pkg/monitortests/network/legacynetworkmonitortests/networking.go [hunk 6]
```diff
@@ -21,6 +23,32 @@ type testCategorizer struct {
...
+	kubeClient, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return platform, fmt.Errorf("error creating kubeClient: %v", err)
+	}
+	isMicroShift, err := exutil.IsMicroShiftCluster(kubeClient)
+	if err != nil {
+		return platform, fmt.Errorf("error checking MicroShift cluster: %v", err)
+	}
+	if isMicroShift {
+		return platform, nil
```
Comment (deads2k) [inline pkg/monitortests/network/legacynetworkmonitortests/networking.go:38]: this will be empty, not nil, correct?  You sure that's what you want?

File: pkg/monitortests/node/kubeletlogcollector/monitortest.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (deads2k) [reply pkg/monitortests/node/kubeletlogcollector/monitortest.go:34]: Sounds like you need to fix this code then, not simply skip it.

**Comment by deads2k** [comment]
This has made obvious, but will not address shortcomings for addressing CI failures.  I'm willing to allow it, but microshift failure resolution will suffer for lack of this.

/approve
/hold

holding for @pacevedom to acknowledge the shortcoming.  You may release on acknowledgement without a fix if you wish.

File: pkg/monitortests/network/legacynetworkmonitortests/networking.go (hunk 6 above)
Comment (pacevedom) [reply pkg/monitortests/network/legacynetworkmonitortests/networking.go:38]: Yes, the platform is only needed when deploying on Azure, and thats not the case for MicroShift.

**Comment by pacevedom** [comment]
As we already talked about it over Slack, posting here a summary.
The shortcomings account for addressing failures in CI and how to debug them. Since MicroShift does not work/is not deployed like OpenShift there are a few key differences.
In this case, for example, the logs for kubelet and other components are to be found in the journal for microshift itself, not in /var/log/ which is where one of the tests is looking.
To address these limitations, a step in CI is included to run an sos report including microshift specific plugins.
Will work in the future to try to overcome some of these limitations where it makes sense.

/hold cancel because of the reasons above.
